src/deployment/deployment-hapi-plugin.ts

```diff
@@ -129,24 +129,27 @@ class DeploymentHapiPlugin extends HapiPlugin {
 
   protected parseHostPre(request: Hapi.Request, reply: Hapi.IReply) {
     const key = getDeploymentKeyFromHost(request.app.hostname);
-
     if (!key) {
       return reply(Boom.create(
         403,
         `Could not parse deployment URL from hostname '${request.app.hostname}'`,
       ));
     }
     return reply(key);
-
   }
 
   protected async authorizePre(request: Hapi.Request, reply: Hapi.IReply) {
     try {
       const projectId: number = request.pre[PREKEY].projectId;
       const deploymentId: number = request.pre[PREKEY].deploymentId;
+      console.log(projectId);
+      console.log(deploymentId);
       if (await request.userHasAccessToDeployment(projectId, deploymentId, request.auth.credentials)) {
         return reply('ok');
       }
+      if (request.isInternal) {
+        return reply('ok');
+      }
```

Comment:
Should this be done first to avoid the unnecessary `userHasAccessToDeployment` lookup?